american-geophysical-union.csl

```diff
@@ -126,109 +471,226 @@
+      <else-if type="bill legal_case legislation" match="none">
         <text variable="container-title" font-style="italic"/>
-      </group>
-      <group delimiter=" ">
-        <text term="volume" form="short"/>
-        <text variable="volume"/>
-      </group>
-      <text macro="editor"/>
-    </group>
+      </else-if>
+    </choose>
+  </macro>
+  <macro name="legal-cites">
+    <choose>
+      <if type="legal_case" match="any">
+        <group prefix=", " delimiter=" ">
+          <choose>
+            <if variable="container-title">
+              <text variable="volume"/>
+              <text variable="container-title"/>
+              <group delimiter=" ">
+                <!--change to label variable="section" as that becomes available -->
+                <text term="section" form="symbol"/>
+                <text variable="section"/>
+              </group>
+              <text variable="page"/>
+            </if>
+            <else>
+              <text variable="number" prefix="No. "/>
+            </else>
+          </choose>
+        </group>
+      </if>
+      <else-if type="bill legislation" match="any">
+        <group delimiter=", " prefix=", ">
+          <choose>
+            <if variable="number">
+              <!--There's a public law number-->
+              <text variable="number" prefix="Pub. L. No. "/>
+              <group delimiter=" ">
+                <!--change to label variable="section" as that becomes available -->
+                <text term="section" form="symbol"/>
+                <text variable="section"/>
+              </group>
+              <group delimiter=" ">
+                <text variable="volume"/>
+                <text variable="container-title"/>
+                <text variable="page-first"/>
+              </group>
+            </if>
+            <else>
+              <group delimiter=" ">
+                <text variable="volume"/>
+                <text variable="container-title"/>
+                <!--change to label variable="section" as that becomes available -->
+                <text term="section" form="symbol"/>
+                <text variable="section"/>
+              </group>
+            </else>
+          </choose>
+        </group>
+      </else-if>
+    </choose>
   </macro>
-  <citation et-al-min="3" et-al-use-first="1" disambiguate-add-year-suffix="true" collapse="year">
+  <macro name="original-date">
+    <choose>
+      <if variable="original-date">
+        <group prefix="(" suffix=")" delimiter=" ">
+          <!---This should be localized-->
+          <text value="Original work published"/>
+          <date variable="original-date" form="text"/>
+        </group>
+      </if>
+    </choose>
+  </macro>
+  <citation et-al-min="3" et-al-use-first="1" disambiguate-add-year-suffix="true" disambiguate-add-names="true" disambiguate-add-givenname="true" collapse="year" givenname-disambiguation-rule="primary-name">
     <sort>
-      <key macro="year-date"/>
```

Comment:
I think they kept their old sorting rules:


    First author alone, chronologically (most recent first):

Smith, R. (2000a). ….
Smith, R. (2000b). ….
Smith, R. (2003). ….

    With one coauthor, alphabetically by coauthor and then chronologically:

Smith, R., & Frank, L. A. (1998). ….
Smith, R., & Allen, F. A. (2001). ….
Smith, R., & Frank, L. A. (2001). ….

    With two or more coauthors, chronologically only:

Smith, R., & Roberts, D. H. (2005). ….
Smith, R., Roberts, D. H., & Jones, J. (1998). ….
Smith, R., Allen, F. A., & Baker, T. L. (1999). ….
Smith, T. (1998). ….